aws_s3_rules/aws_s3_access_error.py

```diff
@@ -8,6 +8,9 @@
 
 
 def rule(event):
+    if event.get('useragent', '').startswith('aws-internal'):
```

Comment:
User agent's are usually pretty easily spoofable. Is there a field that has higher fidelity here? Based off a quick check of the some recent events it appears that all the requests with user agent `aws-internal` have private IP addresses (RFC 1918), which should be impossible from a non-AWS system. My only concern is I'm not sure if it's possible with e.g. VPC endpoints or something to communicate to s3 straight from a private IP address. All of our systems (log processor etc.) appear to be coming from public IPs.

Reply:
For what it's worth, AWS publishes their ranges here: https://ip-ranges.amazonaws.com/ip-ranges.json